server/game/gamesteps/EventAggregator

```diff
@@ -0,0 +1,9 @@
+class EventAggregator {
+    constructor(eventName, name, compareValFunc) {
+        this.eventName = eventName;
+        this.name = name || 'aggregate';
+        this.compareValFunc = compareValFunc || (() => true);
```

Comment:
I'd recommend renaming this (and the parameter) to be `groupBy` or `aggregateBy` instead. When aggregating, you're using the result of this function to group events together rather than merely comparing them.